src/main/java/services/RestoreFactory.java

```diff
@@ -159,8 +161,20 @@ public UserSqlSandbox performTimedSync(boolean shouldPurge) throws Exception {
         if(getSqlSandbox().getLoggedInUser() != null){
             getSQLiteDB().createDatabaseFolder();
         }
-        UserSqlSandbox sandbox = restoreUser();
-        if (shouldPurge) {
+        UserSqlSandbox sandbox;
+        try {
+             sandbox = restoreUser();
+        } catch (HttpClientErrorException e) {
+            if (e.getStatusCode().value() == 412) {
+                getSQLiteDB().deleteDatabaseFile();
+                // this line has the effect of clearing the sync token
+                // from the restore URL that's used
+                sqLiteDB = new UserDB(domain, username, asUsername);
```

Comment:
@ctsims @shubham1g5 My testing cycle locally was just
1. Delete synclogs to force a 412 (using this new button: https://github.com/dimagi/commcare-hq/pull/25850)
2. Go to WebApps and hit sync

Previously the behavior was that it raised an error and prevented you from syncing again. Now it syncs seamlessly.

Do these two lines seem like a reasonable way of clearing user state in before resyncing in response to a 412? On a phone there may be unsent forms, but I think WebApps doesn't do unsent forms, just incomplete forms. Does that sound right? Anything else I'm overlooking?

Reply:
Hi Danny,

On device we do this with an atomic transaction (IE: Try to download the sync, and only clear user data once it's received in full, and if the download fails don't break the state), but I agree that  WebApps doesn't need that level of care and we generally consider it to be a safe action to wipe out someone's user sandbox and force them to resync (it's how we handle ~all migrations).

I might double check that there's no utility method somewhere which wraps 
```
getSQLiteDB().deleteDatabaseFile();
```
in a more semantically meaningful action which might come with some other cleanup or considerations, but the approach itself seems fine